kartograf/merge.py

```diff
@@ -67,24 +84,34 @@ def general_merge(base_file, extra_file, extra_filtered_file, out_file, silent=F
             extra_nets_int.append(netw_int)
             extra_asns.append(asn.strip())
             extra_pfxs.append(pfx)
+            if ipn.version == 4:
+                root_net = int(pfx.split('.')[0])
+            else:
+                root_net = str(pfx).split(':')[0]
+            extra_pfxs_leading.append(root_net)
 
     df_extra = pd.DataFrame()
     df_extra['INETS'] = extra_nets_int
     df_extra['ASNS'] = extra_asns
     df_extra['PFXS'] = extra_pfxs
+    df_extra['PFXS_LEADING'] = extra_pfxs_leading
 
-    def check_inclusion(extra_net):
-        inclusion_list = int(extra_net) & net_masks == network_addresses
-
-        if np.any(inclusion_list):
-            return 1
+    print("Merging extra prefixes that were not "
+          "included in the base file:\n")
 
-        return 0
+    included = []
+    for row in df_extra.itertuples(index=False):
```

Comment:
for each row in the dataframe, lookup its leading prefix (leftmost section) in the base_dict. Pass the row and the dict entry (the subset of IPs and net masks to check). If its not in the base_dict, its not included so return 0.